README.md

```diff
@@ -1,27 +1,508 @@
+
+Instead we typically want to dynamically load the data from some API endpoint,
+for instance `GET /api/posts`. So whenever we render a new blog component, we
+do a XHR call and load the blog posts that we want to render.
+
+In React, the function that gets executed once a component has been "rendered"
+(= created) is called `componentDidMount`. `componentDidMount` is a lifecycle
+method, as such it is **not** available for pure components (such as the
+`BlogPost` component that we defined using an arrow function):
+
+```js
+class Blog extends React.Component {
+    componentDidMount () {
+        fetch('http://localhost:4000/api/posts')
+            .then(response => response.json())
+            .then(posts => {
+                console.log('Yey! I got some posts!', posts)
+            })
+    }
+    render () {
+        return (
+            <div>
+                <BlogPost title={'Hello World!'} body={'Hello, this is my blog.'} />
+                <BlogPost title={'Good bye!'} body={'I\'m busy. I\'m shutting this blog down.'} />
+            </div>
+        )
+    }
+}
+```
+
+This means that whenever we `mount` this component by rendering it as a child of
+some other part of our application, e.g. using `<Blog />`, React executes the
+`componentDidMount` method that requests data from our API.
+
+Once we receive those blog posts, we `console.log` them. While
+`console.log`ging them is certainly fun, our users won't be too happy about
+having to open the Chrome DevTools in order to read our blog. So instead of just
+printing them to the console, the `Blog` component should somehow "keep track
+of them".
+
+The most intuitive solution would be to "set" them as a property on the
+component. React provides a method for that called... `setState`! `setState`
+allows components to keep track of their **internal** state.
+
+So let's change our component to update the state of the blog component once we
+receive the requested blog posts from our API:
+
+```js
+    componentDidMount () {
+        fetch('http://localhost:4000/api/posts')
+            .then(response => response.json())
+            .then(posts => {
+                console.log('Yey! I got some posts!', posts)
+
+                // Let's update the state here:
+                this.setState({ posts: posts })
+
+                // or shorter: this.setState({ posts })
+            })
+    }
+```
+
+Now that we updated the component's state using `setState`, we want to render
+our posts using the `BlogPost` component, so we need to update our component's
+render method:
+
+```js
+    render () {
+        return (
+            <div>
+                {
+                    this.state.posts.map((post) =>
+                        <BlogPost title={post.title} body={post.body} />
+                    )
+                }
+            </div>
+        )
+    }
+```
+
+Keep in mind that we didn't change the `BlogPost` component *at all* — ideally
```

Comment:
This is great — it sort of alludes to pure components but doesn't talk about them specifically yet (this is for a later lesson)